examples/all-clusters-app/ambd/main/DeviceCallbacks.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file DeviceCallbacks.cpp
+ *
+ * Implements all the callbacks to the application from the CHIP Stack
+ *
+ **/
+#include "DeviceCallbacks.h"
+
+#include "CHIPDeviceManager.h"
+#include <app/util/af.h>
+#include <app/common/gen/attribute-id.h>
+#include <app/common/gen/cluster-id.h>
+#include <app/Command.h>
+#include <app/server/Mdns.h>
+#include <app/util/basic-types.h>
+#include <app/util/util.h>
+#include <lib/mdns/Advertiser.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+#include <support/logging/Constants.h>
+
+static const char * TAG = "app-devicecallbacks";
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::System;
+using namespace ::chip::DeviceLayer;
+using namespace ::chip::DeviceManager;
+using namespace ::chip::Logging;
+
+uint32_t identifyTimerCount;
+constexpr uint32_t kIdentifyTimerDelayMS = 250;
+
+void DeviceCallbacks::DeviceEventCallback(const ChipDeviceEvent * event, intptr_t arg)
+{
+    //TODO
+}
+
+void DeviceCallbacks::PostAttributeChangeCallback(EndpointId endpointId, ClusterId clusterId, AttributeId attributeId, uint8_t mask,
+                                                  uint16_t manufacturerCode, uint8_t type, uint16_t size, uint8_t * value)
+{
+    printf("[%s] PostAttributeChangeCallback - Cluster ID: 0x%04x, EndPoint ID: 0x%02x, Attribute ID: 0x%04x\r\n", TAG, clusterId,
+                  endpointId, attributeId);
+
+    switch (clusterId)
+    {
+    case ZCL_IDENTIFY_CLUSTER_ID:
+        OnIdentifyPostAttributeChangeCallback(endpointId, attributeId, value);
+        break;
+
+    default:
+        printf("[%s] Unhandled cluster ID:  0x%04x\r\n", TAG, clusterId);
+        break;
+    }
+}
+
+void IdentifyTimerHandler(Layer * systemLayer, void * appState, Error error)
+{
+    if (identifyTimerCount)
+    {
+        identifyTimerCount--;
+    }
+}
+
+void DeviceCallbacks::OnIdentifyPostAttributeChangeCallback(EndpointId endpointId, AttributeId attributeId, uint8_t * value)
+{
+    VerifyOrExit(attributeId == ZCL_IDENTIFY_TIME_ATTRIBUTE_ID, printf ("[%s] Unhandled Attribute ID: '0x%04x", TAG, attributeId));
```

Comment:
consider using simple if statements to printf and return to make the code easier to read.

Reply:
This line of code actually refers to the writing of other platforms.